From 808bfce08a07c62d1e46d1d53386c6578165ca6c Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 28 Apr 2022 17:09:37 +0000
Subject: GCC: fix ambiguous references in QueryProcessor

GCC does not allow aliases with same name. Use explicit namespace to
avoid confusion.

Bug: 819294
Change-Id: Icc5a406bd7c51d928118b5805e6bb024b383c490
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3602372
Reviewed-by: Siddhartha S <ssid@chromium.org>
Reviewed-by: Shakti Sahu <shaktisahu@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#997251}
---
 .../internal/execution/processing/query_processor.h         | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/components/segmentation_platform/internal/execution/processing/query_processor.h b/components/segmentation_platform/internal/execution/processing/query_processor.h
index 403446b0dbf1c..dc3366a0f14c4 100644
--- a/components/segmentation_platform/internal/execution/processing/query_processor.h
+++ b/components/segmentation_platform/internal/execution/processing/query_processor.h
@@ -22,9 +22,9 @@ class QueryProcessor {
  public:
   // TODO(ssid): Remove these indirections and replace all uses with the global
   // types.
-  using Tensor = Tensor;
-  using IndexedTensors = IndexedTensors;
-  using FeatureIndex = FeatureIndex;
+  using Tensor = segmentation_platform::processing::Tensor;
+  using IndexedTensors = segmentation_platform::processing::IndexedTensors;
+  using FeatureIndex = segmentation_platform::processing::FeatureIndex;
 
   // TODO(haileywang): Maybe use a unique_ptr<> here.
   using QueryProcessorCallback =
-- 
2.32.0

